src/misc/setup.cpp

```diff
@@ -415,6 +402,21 @@ bool Prop_string::IsValidValue(const Value& in)
 		return true;
 	}
 
+	// If the property's valid values includes either positive or negative
+	// bool strings ("on", "false", etc..), then check if this incoming
+	// value is either.
+	if (is_positive_bool_valid || is_negative_bool_valid) {
+		const auto is_bool = parse_bool_setting(in.ToString());
+		if (is_bool) {
+			if (*is_bool == true && is_positive_bool_valid) {
+				return true;
+			}
+			if (*is_bool == false && is_negative_bool_valid) {
+				return true;
+			}
+		}
+	}
```

Comment:
This is a bit "inside out" 😅 Could "invert" it and make it a lot simpler by using a `valid_bool_value` lambda, something like this:

```cpp
if (is_positive_bool_valid && valid_bool_value(in, true)) {
   return true;
} else if (is_negative_bool_valid && valid_bool_value(in, false)) {
   return false;
}
```